modules/dynamic/src/smithy4s/dynamic/DynamicModel.scala

```diff
@@ -0,0 +1,18 @@
+package smithy4s
+package dynamic
+
+class DynamicModel(
```

Comment:
TODO : we'll need to 

* have the public methods of this construct return existential versions (honestly the allServices is gonna be a pita on Scala 2 ...) 
* move most things that leads to the construction in an `internals` package

Reply:
First point is done : I used abstract type members to circumvent the lack of higher-kinded existentials in Scala 2. That hides the `Any` from the end user, forcing them to only use dynamic services as "proxies"